src/controller_examples/zookeeper_controller/proof/liveness.rs

```diff
@@ -1217,17 +922,129 @@ proof fn lemma_true_leads_to_always_current_state_matches_zk_under_eventual_inva
         spec,
         true_pred(),
         lift_state(|s: ClusterState| !s.reconcile_state_contains(zk.object_ref())),
-        lift_state(at_init_step_with_no_pending_req(zk)),
-        lift_state(at_after_create_headless_service_step_with_zk_and_pending_req_in_flight(zk)),
-        lift_state(at_after_create_client_service_step_with_zk_and_pending_req_in_flight(zk)),
-        lift_state(at_after_create_admin_server_service_step_with_zk_and_pending_req_in_flight(zk)),
-        lift_state(at_after_create_config_map_step_with_zk_and_pending_req_in_flight(zk)),
-        lift_state(at_after_get_stateful_set_step_with_zk_and_pending_req_in_flight(zk)),
+        lift_state(no_pending_req_at_zookeeper_step_with_zk(zk, ZookeeperReconcileStep::Init)),
+        lift_state(pending_req_in_flight_at_zookeeper_step_with_zk(ZookeeperReconcileStep::AfterCreateHeadlessService, zk, arbitrary())),
+        lift_state(pending_req_in_flight_at_zookeeper_step_with_zk(ZookeeperReconcileStep::AfterCreateClientService, zk, arbitrary())),
+        lift_state(pending_req_in_flight_at_zookeeper_step_with_zk(ZookeeperReconcileStep::AfterCreateAdminServerService, zk, arbitrary())),
+        lift_state(pending_req_in_flight_at_zookeeper_step_with_zk(ZookeeperReconcileStep::AfterCreateConfigMap, zk, arbitrary())),
+        lift_state(pending_req_in_flight_at_zookeeper_step_with_zk(ZookeeperReconcileStep::AfterGetStatefulSet, zk, arbitrary())),
         lift_state(current_state_matches(zk)),
         always(lift_state(current_state_matches(zk)))
     );
 }
 
+// This lemma ensures that zookeeper controller at some step with pending request in flight will finally enter its next step.
```

Comment:
To avoid ambiguity, you can say "This lemma ensures that zookeeper controller at step a with pending request in flight will finally enter step b." and use step a, step b below, instead of initial, final step. My first reaction when seeing initial step is actually `ZookeeperReconcileStep::Init`